### PR TITLE
fix: deduplicate message properties, message descriptors

### LINKED ISSUE #69: Invalid proto message gets generated when using anyOf, oneOf, allOf
Given schema:
```
components:
  schemas:
    TestOneOfApiResponse:
      oneOf:
        - $ref: "#/components/schemas/Person"
        - $ref: "#/components/schemas/Order"
    Person:
      properties:
        id:
          type: integer
       name:
          type: string
    Order:
      properties:
        id:
          type: integer
```
currently results in the following proto file:
```
message TestOneOfApiResponse {
  int32 id = 1;

  string name = 2;

  int32 id = 3;
}
```

`TestOneOfApiResponse` is not a valid message, because `id` exists twice.

### PR DESCRIPTION
**Pull request by TylerGillson**
- fixes #69
- fixes a previously unreported issue where duplicate message descriptors are generated for POST requests with an object ref in their body

### CODE REVIEW

File: generator/generator_service.go [hunk 1]
```diff
@@ -129,6 +129,12 @@ func buildInputTypeAndOutputType(parametersTypeName string, responseTypeName str
 	if responseTypeName == "" {
 		outputType = "google.protobuf.Empty"
 	}
+	for _, t := range types {
```
Comment (LorenzHW) [inline generator/generator_service.go:132]: Does this change affect the integration tests?

**Review by LorenzHW — COMMENTED** [review]
@TylerGillson Thanks for the contribution! Overall LGTM. I just wonder whether you ran the tests locally once? I actually don't remember whether I set that up . I think we should add a GitHub action to run the tests on PR and merge into main. I create a separate issue for this.

File: generator/generator_service.go (hunk 1 above)
Comment (LorenzHW) [reply generator/generator_service.go:132]: https://github.com/google/gnostic-grpc/blob/main/generator/renderer_test.go

Comment (TylerGillson) [reply generator/generator_service.go:132]: @LorenzHW all of the tests that were originally passing are still passing. I've fixed the GH action [here](https://github.com/google/gnostic-grpc/pull/91).

I think this test failure:
> 2023/06/06 10:23:55 The query parameter with the Name param2 is invalid. Note that fields which are mapped to URL query parameters must have a primitive type or a repeated primitive type or a non-repeated message type. See: https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L118 for more information.

Has always been an issue, based on the TODO in parameters.yaml.

Comment (LorenzHW) [reply generator/generator_service.go:132]: Thanks @TylerGillson. I think at some point the tests were green but it is ok for now. Still haven't had the chance to take a look at the repository but it is high on my TODO list...

**Review by LorenzHW — APPROVED** [review]

File: generator/generator_service.go (hunk 1 above)
Comment (TylerGillson) [reply generator/generator_service.go:132]: No problem, happy to help.

**Pull request merged**